c2rust-analyze/src/context.rs

```diff
@@ -186,6 +186,59 @@ bitflags! {
     }
 }
 
+bitflags! {
+    /// Flags indicating reasons why a function isn't being rewritten.
+    #[derive(Default)]
+    pub struct DontRewriteFnReason: u16 {
+        /// The user requested that this function be left unchanged.
+        const USER_REQUEST = 0x0001;
+        /// The function contains an unsupported int-to-pointer cast.
+        const INT_TO_PTR_CAST = 0x0002;
+        /// The function calls an extern function that has no safe rewrite.
+        const EXTERN_CALL = 0x0004;
+        /// The function calls another local function that isn't being rewritten.
+        const NON_REWRITTEN_CALLEE = 0x0008;
+        /// The function uses `Cell` pointers whose types are too complex for the current rewrite
+        /// rules.
+        const COMPLEX_CELL = 0x0010;
+        /// The function contains a pointer-to-pointer cast that isn't covered by rewrite rules.
+        const PTR_TO_PTR_CAST = 0x0020;
+        /// The function dereferences a pointer that would remain raw after rewriting.
+        const RAW_PTR_DEREF = 0x0040;
+
+        /// Dataflow analysis failed on this function.
+        const DATAFLOW_FAILED = 0x0080;
+        /// Borrowcheck/Polonius analysis failed on this function.
+        const BORROWCK_FAILED = 0x0100;
+        /// MIR rewrite generation failed on this function.
+        const MIR_REWRITE_FAILED = 0x0200;
+        /// HIR/AST rewrite generation failed on this function.
+        const HIR_REWRITE_FAILED = 0x0200;
+    }
+}
+
+bitflags! {
+    /// Flags indicating reasons why a static isn't being rewritten.
+    #[derive(Default)]
+    pub struct DontRewriteStaticReason: u16 {
+        /// The user requested that this static be left unchanged.
+        const USER_REQUEST = 0x0001;
+        /// The field is used in a function that isn't being rewritten.
+        const NON_REWRITTEN_USER = 0x0002;
```

Comment:
Nit: I'd call this `_USAGE` or `_USE` to distinguish the senses of "user" here. Not a blocker, of course.

Reply:
Good idea - I renamed both to `_USE`